pkg/apis/mongodb/v1/mongodb_types.go

```diff
@@ -42,6 +45,35 @@ type MongoDBSpec struct {
 	// be set for the deployment
 	// +optional
 	FeatureCompatibilityVersion string `json:"featureCompatibilityVersion,omitempty"`
+
+	// Security configures security features, such as TLS, for a deployment
+	// +optional
+	Security Security `json:"security"`
+}
+
+type Security struct {
+	// TLS configuration for both client-server and server-server communication
+	// +optional
+	TLS TLS `json:"tls"`
+}
+
+// TLS is the configuration used to set up TLS encryption
+type TLS struct {
+	Enabled bool `json:"enabled"`
+
+	// Optional configures if TLS should be required or optional for connections
+	// +optional
+	Optional bool `json:"optional"`
+
+	// ServerSecretName is the name of a secret containing a private key and certificate to use for TLS
+	// The key and cert are expected to be PEM encoded and available at "tls.key" and "tls.crt"
+	// +optional
+	ServerSecretName string `json:"serverSecretName"`
+
+	// CAConfigMapName is the name of a ConfigMap containing the certificate for the CA which signed the server certificates
+	// The certificate is expected to be available under the key "ca.crt"
+	// +optional
+	CAConfigMapName string `json:"caConfigMapName"`
```

Comment:
This one is better than the Server, because it implies something related to the CA.

At the same time we have `corev1.ConfigMapKeySelector` and `corev1.SecretKeySelector` that can very nicely support some of our use-cases, like, not making the users use a hard-coded ConfigMap entry name, but any name they like.

Reply:
Didn't know about those types, thanks! We should definitely look into using those (or something similar) to improve flexibility.